backend/icpy/api/rest_api.py

```
@@ -645,23 +647,115 @@ async def get_file_raw(path: str, namespace: Optional[str] = None):
                 if abs_path is not None:
                     logger.info(f"[REST][raw] Using local streaming path abs={abs_path}")
                     # Security: ensure selected path is inside either fs_root or workspace_root
-                    if not (abs_path.startswith(fs_root) or abs_path.startswith(workspace_root)):
+                    try:
+                        abs_path_obj = Path(abs_path).resolve()
+                        fs_root_obj = Path(fs_root).resolve()
+                        workspace_root_obj = Path(workspace_root).resolve()
+                        try:
+                            abs_path_obj.relative_to(fs_root_obj)
+                        except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)